### ISSUE TITLE
Printer Not Ready

### ISSUE THREAD

**Issue by WarriorClub** [KevinOConnor/klipper#3223]
Hi,

I am fairly new to Klipper and 3D printing as a whole.  I am trying to get Klipper to run through a Raspberry Pi4 which connects to Octoprint and I have configured everything as stated on the guide.
However I keep getting Printer not ready when I type status in Octoprint.
I have found the the serial ID through Putty with the "ls /dev/serial/by-id/*" Which came out as
"/dev/serial/by-id/usb-Klipper_stm32f100xe_35FFD6054246303613811057-if00"
But no luck. Still keeps saying not ready.  Any suggestions would be helpful.

Here is the Klippy log.

# Automatically generated file; DO NOT EDIT.
# Klipper Firmware Configuration
#
CONFIG_LOW_LEVEL_OPTIONS=y
# CONFIG_MACH_AVR is not set
# CONFIG_MACH_ATSAM is not set
# CONFIG_MACH_ATSAMD is not set
# CONFIG_MACH_LPC176X is not set
CONFIG_MACH_STM32=y
# CONFIG_MACH_PRU is not set
# CONFIG_MACH_LINUX is not set
# CONFIG_MACH_SIMU is not set
CONFIG_STEP_DELAY=2
CONFIG_BOARD_DIRECTORY="stm32"
CONFIG_MCU="stm32f103xe"
CONFIG_CLOCK_FREQ=72000000
CONFIG_USBSERIAL=y
CONFIG_SERIAL_PORT=1
CONFIG_FLASH_START=0x8007000
CONFIG_FLASH_SIZE=0x10000
CONFIG_RAM_START=0x20000000
CONFIG_RAM_SIZE=0x5000
CONFIG_STACK_SIZE=512
CONFIG_STM32_SELECT=y
CONFIG_MACH_STM32F103=y
# CONFIG_MACH_STM32F207 is not set
# CONFIG_MACH_STM32F405 is not set
# CONFIG_MACH_STM32F407 is not set
# CONFIG_MACH_STM32F446 is not set
# CONFIG_MACH_STM32F042 is not set
# CONFIG_MACH_STM32F070 is not set
CONFIG_MACH_STM32F1=y
CONFIG_HAVE_STM32_USBFS=y
# CONFIG_HAVE_STM32_USBOTG is not set
# CONFIG_STM32_FLASH_START_800 is not set
# CONFIG_STM32_FLASH_START_2000 is not set
# CONFIG_STM32_FLASH_START_5000 is not set
CONFIG_STM32_FLASH_START_7000=y
# CONFIG_STM32_FLASH_START_10000 is not set
# CONFIG_STM32_FLASH_START_0000 is not set
# CONFIG_ARMCM_RAM_VECTORTABLE is not set
CONFIG_STM32_CLOCK_REF_8M=y
# CONFIG_STM32_CLOCK_REF_12M is not set
# CONFIG_STM32_CLOCK_REF_16M is not set
# CONFIG_STM32_CLOCK_REF_INTERNAL is not set
CONFIG_CLOCK_REF_FREQ=8000000
CONFIG_USB_VENDOR_ID=0x1d50
CONFIG_USB_DEVICE_ID=0x614e
CONFIG_USB_SERIAL_NUMBER_CHIPID=y
CONFIG_USB_SERIAL_NUMBER="12345"

#
# USB ids
#
# CONFIG_CUSTOM_STEP_DELAY is not set
CONFIG_INITIAL_PINS="!PC13"
CONFIG_HAVE_GPIO=y
CONFIG_HAVE_GPIO_ADC=y
CONFIG_HAVE_GPIO_SPI=y
CONFIG_HAVE_GPIO_I2C=y
# CONFIG_HAVE_GPIO_HARD_PWM is not set
CONFIG_HAVE_GPIO_BITBANGING=y
CONFIG_HAVE_STRICT_TIMING=y
CONFIG_HAVE_CHIPID=y
CONFIG_INLINE_STEPPER_HACK=y
=======================
Build file /home/pi/klipper/klippy/../out/klipper.dict(5569): Fri Aug 21 15:59:40 2020
Last MCU build version: v0.8.0-721-g560d48db-dirty-20200821_155940-octopi
Last MCU build tools: gcc: (15:7-2018-q2-6) 7.3.1 20180622 (release) [ARM/embedded-7-branch revision 261907] binutils: (2.31.1-11+rpi1+11) 2.31.1
Last MCU build config: BUS_PINS_i2c1=PB6,PB7 BUS_PINS_i2c2=PB10,PB11 STEP_DELAY=2 INITIAL_PINS=!PC13 ADC_MAX=4095 BUS_PINS_spi3=PB4,PB5,PB3 BUS_PINS_spi2=PB14,PB15,PB13 BUS_PINS_spi1=PA6,PA7,PA5 RESERVE_PINS_USB=PA11,PA12 MCU=stm32f103xe STATS_SUMSQ_BASE=256 CLOCK_FREQ=72000000
Build file /home/pi/klipper/klippy/../out/klipper.elf(99488): Fri Aug 21 15:59:44 2020
Restarting printer
Start printer at Fri Aug 21 17:06:36 2020 (1598025996.9 1251.5)
===== Config file =====
[stepper_x]
step_pin = PB13
dir_pin = !PB12
enable_pin = !PB14
step_distance = .0125
endstop_pin = ^PC0
position_endstop = 0
position_max = 235
homing_speed = 50

[tmc2209 stepper_x]
uart_pin = PB15
microsteps = 16
run_current = 0.580
hold_current = 0.500
stealthchop_threshold = 250

[stepper_y]
step_pin = PB10
dir_pin = !PB2
enable_pin = !PB11
step_distance = .0125
endstop_pin = ^PC1
position_endstop = 0
position_max = 235
homing_speed = 50

[tmc2209 stepper_y]
uart_pin = PC6
microsteps = 16
run_current = 0.580
hold_current = 0.500
stealthchop_threshold = 250

[stepper_z]
step_pin = PB0
dir_pin = PC5
enable_pin = !PB1
step_distance = .0025
endstop_pin = probe:z_virtual_endstop
position_endstop = 1
position_min = -2
position_max = 250

[tmc2209 stepper_z]
uart_pin = PC10
microsteps = 16
run_current = 0.580
hold_current = 0.500
stealthchop_threshold = 5

[extruder]
step_pin = PB3
dir_pin = !PB4
enable_pin = !PD2
step_distance = 0.010526
nozzle_diameter = 0.400
filament_diameter = 1.750
heater_pin = PC8
sensor_type = EPCOS 100K B57560G104F
sensor_pin = PA0
control = pid
pid_kp = 21.527
pid_ki = 1.063
pid_kd = 108.982
min_temp = 0
max_temp = 250

[tmc2209 extruder]
uart_pin = PC11
microsteps = 16
run_current = 0.650
hold_current = 0.500
stealthchop_threshold = 5

[heater_bed]
heater_pin = PC9
sensor_type = ATC Semitec 104GT-2
sensor_pin = PC3
control = pid
pid_kp = 54.027
pid_ki = 0.770
pid_kd = 948.182
min_temp = 0
max_temp = 130

[fan]
pin = PA8

[mcu]
serial = /dev/serial/by-id/usb-Klipper_stm32f100xe_35FFD6054246303613811057-if00

[printer]
kinematics = cartesian
max_velocity = 300
max_accel = 3000
max_z_velocity = 5
max_z_accel = 100

[static_digital_output usb_pullup_enable]
pins = !PC13

[board_pins]
aliases = 
	EXP1_1=PB5, EXP1_3=PA9,   EXP1_5=PA10, EXP1_7=PB8, EXP1_9=<GND>,
	EXP1_2=PB6, EXP1_4=<RST>, EXP1_6=PB9,  EXP1_8=PB7, EXP1_10=<5V>

[display]
lcd_type = st7920
cs_pin = EXP1_7
sclk_pin = EXP1_6
sid_pin = EXP1_8
encoder_pins = ^EXP1_5, ^EXP1_3
click_pin = ^!EXP1_2

[output_pin beeper]
pin = EXP1_1

[bltouch]
sensor_pin = ^PC2
control_pin = PA1
x_offset = -40
y_offset = -10
z_offset = 0

[bed_mesh]
speed = 50
horizontal_move_z = 5
mesh_min = 44, 44
mesh_max = 191, 191
probe_count = 3, 3
mesh_pps = 2,2

[gcode_macro G29]
gcode = 
	BED_MESH_CLEAR
	BED_MESH_CALIBRATE
	BED_MESH_OUTPUT

[safe_z_home]
home_xy_position = 155,120
speed = 80.0
z_hop = 10.0
z_hop_speed = 10.0
=======================
Extruder max_extrude_ratio=0.266081
Starting serial connect
Unable to open port: [Errno 2] could not open port /dev/serial/by-id/usb-Klipper_stm32f100xe_35FFD6054246303613811057-if00: [Errno 2] No such file or directory: '/dev/serial/by-id/usb-Klipper_stm32f100xe_35FFD6054246303613811057-if00'

Printer is not ready
The klippy host software is attempting to connect.  Please
retry in a few moments.
Klipper state: Not ready
Unable to open port: [Errno 2] could not open port /dev/serial/by-id/usb-Klipper_stm32f100xe_35FFD6054246303613811057-if00: [Errno 2] No such file or directory: '/dev/serial/by-id/usb-Klipper_stm32f100xe_35FFD6054246303613811057-if00'
Unable to open port: [Errno 2] could not open port /dev/serial/by-id/usb-Klipper_stm32f100xe_35FFD6054246303613811057-if00: [Errno 2] No such file or directory: '/dev/serial/by-id/usb-Klipper_stm32f100xe_35FFD6054246303613811057-if00'
Unable to open port: [Errno 2] could not open port /dev/serial/by-id/usb-Klipper_stm32f100xe_35FFD6054246303613811057-if00: [Errno 2] No such file or directory: '/dev/serial/by-id/usb-Klipper_stm32f100xe_35FFD6054246303613811057-if00'
Unable to open port: [Errno 2] could not open port /dev/serial/by-id/usb-Klipper_stm32f100xe_35FFD6054246303613811057-if00: [Errno 2] No such file or directory: '/dev/serial/by-id/usb-Klipper_stm32f100xe_35FFD6054246303613811057-if00'
Unable to open port: [Errno 2] could not open port /dev/serial/by-id/usb-Klipper_stm32f100xe_35FFD6054246303613811057-if00: [Errno 2] No such file or directory: '/dev/serial/by-id/usb-Klipper_stm32f100xe_35FFD6054246303613811057-if00'

Thanks in advance

**Comment by KevinOConnor** [comment]
If you look closely, you'll see that there is a typo in `/dev/serial/by-id/usb-Klipper_stm32f100xe_35FFD6054246303613811057-if00`.  Make sure to copy-and-paste the actual filename as described in the Installation document.

-Kevin

**Comment by WarriorClub** [comment]
Hi Kevin,

Thank you.  I did get it to work on the raspberry pi3B plus.  I guess I did not copy it correctly.  
I have another issue here in trying to get the serial port ID.  I recently got an SKR mini E3 V2.0.  When I try to get it through putty I get ls: cannot access '/dev/serial/by-id/*': No such file or directory
I am using ls /dev/serial/by-id/*

Not sure if this issue is just with me but everything is plugged in.  The raspberry Pi is plugged into the micro SD slot.
